pennylane_qiskit/qiskit_device2.py

```diff
@@ -482,14 +510,19 @@ def _execute_estimator(self, circuit, session):
 
         pauli_observables = [mp_to_pauli(mp, self.num_wires) for mp in circuit.measurements]
         compiled_circuits = self.compile_circuits(qcirc) * len(pauli_observables)
+        estimator.options.update(**self._kwargs)
         # split into one call per measurement
         # could technically be more efficient if there are some observables where we ask
         # for expectation value and variance on the same observable, but spending time on
         # that right now feels excessive
         circ_and_obs = [
             (compiled_circuits[i], pauli_observables[i]) for i in range(len(pauli_observables))
         ]
-        result = estimator.run(circ_and_obs).result()
+        result = estimator.run(
+            circ_and_obs,
+            precision=np.sqrt(1 / circuit.shots.total_shots) if circuit.shots else None,
```

Comment:
Essentially, if you set some kind of precision on device initialization, that will get overwritten if you set some number of shots when running the circuit. Estimator does not take in shots at runtime (only on initialization), but since PL does not take in some `precision` value it doesn't make sense to support `precision` as an option. Therefore, this code just converts a shot value to a precision value, so that the estimator can run the shot value that we set :)

Reply:
How do we come up with this precision value? 🤔